src/main/java/com/labsynch/labseer/service/StandardizationServiceImpl.java

```diff
@@ -0,0 +1,524 @@
+		List<StandardizationDryRunCompound> stndznCompounds = StandardizationDryRunCompound.findStandardizationChanges().getResultList();
+		String json = "[]";
+		if(stndznCompounds.size() > 0) {
+			json = StandardizationDryRunCompound.toJsonArray(stndznCompounds);
+
+		}
+		return(json);
+	}
+	
+	@Override
+	@Transactional
+	public void reset() {
+		boolean truncateTable = chemStructureService.truncateStructureTable(StructureType.DRY_RUN);
+		StandardizationDryRunCompound.truncateTable();
+	}
+	
+	@Override
+	public int restandardizeParentStructures(List<Long> parentIds) throws CmpdRegMolFormatException, StandardizerException, IOException {
+		Parent parent;
+		List<Lot> lots;
+		Lot lot;
+		String originalStructure = null;
+		String result;
+		int totalCount = parentIds.size();
+		logger.info("number of parents to restandardize: " + totalCount);
+		float percent = 0;
+		int p = 1;
+		float previousPercent = percent;
+		previousPercent = percent;
+		for  (Long parentId : parentIds){
+			parent = Parent.findParent(parentId);
+			lots = Lot.findLotsByParent(parent).getResultList();
+			lot = lots.get(0);
+			if (lots.size() > 0 && lot.getAsDrawnStruct() != null){
+				originalStructure = lot.getAsDrawnStruct();				
+			} else {
+				logger.warn("Did not find the asDrawnStruct for parent: " + parentId + "  " + parent.getCorpName());
+				originalStructure = parent.getMolStructure();
+			}
+			
+			result = chemStructureService.standardizeStructure(originalStructure);
+
+			// Compute your percentage.
+			percent = (float)Math.floor(p * 100f / totalCount);
+			if(percent != previousPercent)
+			{
+				// Output if different from the last time.
+				logger.info("standardization " + percent + "% complete");
+			}
+			// Update the percentage.
+			previousPercent = percent;
+			p++;
+
+		}
+		return parentIds.size();
+	}
+
+	@Override
+	public String executeStandardization(String username, String reason) throws StandardizerException{
+		StandardizationHistory standardizationHistory = getMostRecentStandardizationHistory();
+		StandardizerSettingsConfigDTO standardizationSettings = chemStructureService.getStandardizerSettings();
+		if(standardizationHistory == null || StringUtils.equalsIgnoreCase(standardizationHistory.getStandardizationStatus(), "complete")) {
+			standardizationHistory = new StandardizationHistory();
+			standardizationHistory.setSettings(standardizationSettings.toJson());
+			standardizationHistory.setSettingsHash(standardizationSettings.hashCode());
+			standardizationHistory.setRecordedDate(new Date());			
+		}
+		standardizationHistory.setStandardizationStart(new Date());
+		standardizationHistory.setStandardizationComplete(null);
+		standardizationHistory.setStandardizationStatus("running");
+		standardizationHistory.persist();
+		
+		Integer result;
+		try {
+			result = this.runStandardization();
+		} catch (CmpdRegMolFormatException | IOException | StandardizerException e) {
```

Comment:
Does this mean a single bad structure will cause the whole restandardization to fail?
Just want to understand the expected behavior here.

Reply:
Yes, if any one of the exceptions are thrown here, then it will stop and set the standardization to fail.  This of course depends on the underlying code not swallowing any exceptions so I 'll need to make sure they throw when encounting these exceptions.